src/Interactive/EditorFeatures/Core/Extensibility/Interactive/ResetInteractive.cs

```diff
@@ -111,13 +112,21 @@ internal Task Execute(IInteractiveWindow interactiveWindow, string title)
 
             var editorOptions = _editorOptionsFactoryService.GetOptions(interactiveWindow.CurrentLanguageBuffer);
             var importReferencesCommand = referencePaths.Select(_createReference);
-            var importNamespacesCommand = namespacesToImport.Select(_createImport).Join(editorOptions.GetNewLineCharacter());
-            await interactiveWindow.SubmitAsync(importReferencesCommand.Concat(new[]
+            await interactiveWindow.SubmitAsync(importReferencesCommand).ConfigureAwait(true);
+
+            // Project's default namespace might be different from namespace used within project.
+            // Filter out namespace imports that do not exist in interactive compilation.
+            IEnumerable<string> existingNamespaces = await GetNamespacesToImport(namespacesToImport, interactiveWindow).ConfigureAwait(true);
```

Comment:
It would be. But to do this I would need to filter through the namespaces before calling `#r` for all references.